src/modules/pmc/tests/PmcTests.cpp

```diff
@@ -1,42 +1,298 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+#include <fstream>
 #include <gtest/gtest.h>
+#include <vector>
 
 #include <Mmi.h>
 #include <PmcBase.h>
 
-class PmcTests : public PmcBase
+class PmcTestImpl : public PmcBase
 {
 public:
-    PmcTests(unsigned int maxPayloadSizeBytes);
-    ~PmcTests() = default;
+    PmcTestImpl(unsigned int maxPayloadSizeBytes);
+    ~PmcTestImpl() = default;
+    int RunCommand(const char* command, bool replaceEol, std::string* textResult, unsigned int timeoutSeconds) override;
+    void SetTextResult(const std::map<std::string, std::string> &textResults);
 
+private:
+    std::map<std::string, std::string> m_textResults;
 };
 
-PmcTests::PmcTests(unsigned int maxPayloadSizeBytes)
-    : PmcBase(maxPayloadSizeBytes)
+PmcTestImpl::PmcTestImpl(unsigned int maxPayloadSizeBytes)
+: PmcBase(maxPayloadSizeBytes)
 {
 }
 
+void PmcTestImpl::SetTextResult(const std::map<std::string, std::string> &textResults)
+{
+    m_textResults = textResults;
+}
+
+int PmcTestImpl::RunCommand(const char* command, bool replaceEol, std::string* textResult, unsigned int timeoutSeconds)
+{
+    UNUSED(replaceEol);
+    UNUSED(timeoutSeconds);
+
+    std::map<std::string, std::string>::const_iterator it = m_textResults.find(command);
+    if (it != m_textResults.end())
+    {
+        if (it->second == "COMMAND_ERROR")
+        {
+            return 100;
+        }
+        else if (it->second == "TIMEOUT_ERROR")
```

Comment:
Do you convert numeric errors such as ETIME to string ("TIMEOUT_ERROR")? This is not efficient. Use normal numeric values for error codes. Referring to the code under test, not just the unittest

Reply:
Actually, we convert our mocked `TIMEOUT_ERROR` to `ETIME` in order to simulate an error of `RunCommand` method.

Reply:
Why mock as a string instead of the plain ETIME?

Reply:
The goal of the test implementation `RunCommand` function is to mock the expected outcome without actually executing any commands. As such, we have a string-to-string map called `textResults` that specifies as a key the command to be mocked and as a value the expected output of the respective command. We followed the same approach as in the HostName module tests. One thing that is different in our tests, is the fact that we wanted to also simulate errors and that is why we used the hardcoded error strings. Since we are not executing any command, we cannot obtain a real error code, so we simulate the two categories - timeout and non-timeout related command errors. With these special strings (`TIMEOUT_ERROR`, `COMMAND_ERROR`), we were able to simulate both happy and unhappy executions of `RunCommand` using only our string-to-string map and with no additional changes.

To make it clearer, we removed the special strings mentioned above and enhanced the string to string map to a map of string to `tuple<int, string>` as in the example:
  ```
const std::map<std::string, std::tuple<int, std::string>> textResults =
        {
            {"command -v apt-get", std::tuple<int, std::string>(MMI_OK, "")},
            {"command -v apt-get", std::tuple<int, std::string>(ETIME, "")}
        }
```

Let us know if this approach looks better.